### ISSUE TITLE
Scheduler handlers do not take effect in rxjava-2.1.3

### ISSUE THREAD

**Issue by aitorvs** [square/RxIdler#8]
Hi, I am using RxJava 2.1.3 with the latest Espresso 3.0.0 and ATSL 1.0.0.
I noticed that the handlers were not taking effect. I debugged and notice that the call to `Schedulers#initXXScheduler()` happens inside a static block (see `io.reactivex.schedulers.Schedulers:73`).
To get it working, I had to move the calls to `RxJavaPlugins.setInitXXXSchedulerHandler` also inside a static block (in the runner) and then add a `null` check to `DelegatingIdlingResourceScheduler#stopWork` method to make sure `callback` is not `null`.

I am not submitting a PR because I think this fix is somehow a bit dodgy :/ although works for me

Any thoughts?

**Comment by JakeWharton** [comment]
What calls the scheduler inside the runner's onStart method? The app
shouldn't be created until after it returns meaning you have time to
initialize the init function.

**Comment by aitorvs** [comment]
nothing I think. I have some code that unlocks the device and an override `newApplication` that returns the name of my test application to use DI only for test.
`super.onStart()` is the last call.
I also tried to place the calls in `onCreate` rather than `onStart` in the runner with the same result...O_o

**Comment by JakeWharton** [comment]
Yeah sorry I meant onCreate, as that's before the Application is
instantiated.

I'm not quite sure I understand the problem, then, as if nothing references
the schedulers prior to onCreate then you should be able to set the init
function. Do you think you can show some code that reproduces the problem?

**Comment by aitorvs** [comment]
Sure, will try prep something. In the meantimeI was thinking, the schedulers are init inside a [static block](https://github.com/ReactiveX/RxJava/blob/v2.1.1/src/main/java/io/reactivex/schedulers/Schedulers.java#L73). I guess that will be executed when the class is loaded. Any chance of that happening before the runner is able to execute `onCreate()` method??

Does this make sense?

**Comment by aitorvs** [comment]
Update: didn't manage to get 'simple' code that replicates the problem yet but I think I might have some explanation. While debugging I realized that overriding the `newApplication` method in the runner causes the `onCreate()` method in the application to be called before the runner#onCreate() O_o. 
As I create some DI components in the app#onCreate method (one of which creates the Retrofit singleton that adds a `RxJava2CallAdapterFactory` to execute all retrofit2 requests on `io` scheduler), I guess that's the root cause

**Comment by JakeWharton** [comment]
Makes sense. Or well, it doesn't make sense but at least we now know the
cause. We didn't override this method ever. I'll have to see if I can
reproduce and then figure out if it's something we should fix or if there's
a different pattern to solving.

**Comment by aitorvs** [comment]
Don't waste time with that 'update' I gave. I am not sure I found a real
issue. When I was trying to replicate the issue, I did override the
onCreate methods but some of them were just calling super.onCreate() and
the debugger didn't trigger the breakpoint...I guess the compiler optimized
that code out and I though they were not called. I re-tried just and the
runner onCreate() indeed is executed before the app onCreate(). Sorry!
So I am back to square one.
